library/src/G4LatticeLogical.cc

```diff
@@ -690,6 +691,109 @@ void G4LatticeLogical::AddValley(G4double phi, G4double theta, G4double psi) {
   fValleyAxis.push_back(fValleyInv.back()*G4ThreeVector(1.,0.,0.));
 }
 
+// Store drifting-electron valley using valley's direction
+
+void G4LatticeLogical::AddValley(const G4ThreeVector& valleyDirVec, G4bool antival) {
+      
+  // Find the rotation matrix elements. 
+  // The rows correspond to the rotated axis direction 
+  // First row is vx,vy,vz ; second row is a,b,c and third row is d,e,f
+    
+  // We chose the following convention :
+  //  - The rotated y axis must stay in the X-Y plane (c=0)
+  //  - Valley's rotated Z axis must have positive z component (f>0) 
+  //  - Anti-valley's rotated Z axis must have positive z component (f>0) 
+  //  - The three rotated axis should stay normalized :
+  //	vx^2 + vy^2 + vz^2 = 1 ; a^2 + b^2 = 1 ; d^2 + e^2 + f^2 = 1
+  //  - And be orthogonal :
+  //	ad + be = 0 ; avx + bvy = 0 ; dvx + evy + fvz = 0
+  //  - It must be a right hand coordinate system :
+  //	(vx,vy,vz) X (a,b,c) = (d,e,f) : d=-vzb ; e=vza ; f=vxb - vya
```

Comment:
This is fine, but can skip showing d, e, f, expressions.  That comes from the cross-product itself.